stratum/hal/lib/barefoot/bf_pd_wrapper.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020-present Open Networking Founcation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "stratum/hal/lib/barefoot/bf_pd_wrapper.h"
+
+extern "C" {
+#include "tofino/pdfixed/pd_tm.h"
+#include "tofino/pdfixed/pd_devport_mgr.h"
+}
+
+#include "stratum/lib/macros.h"
+
+namespace stratum {
+namespace hal {
+namespace barefoot {
+
+BFPdWrapper* BFPdWrapper::singleton_ = nullptr;
+ABSL_CONST_INIT absl::Mutex BFPdWrapper::init_lock_(absl::kConstInit);
+
+// Gets the CPU port of an unit(device).
+::util::StatusOr<int> BFPdWrapper::GetPcieCpuPort(int unit) {
+  return p4_devport_mgr_pcie_cpu_port_get(unit);
```

Comment:
This one will never return Status failed, right?
The function signature is `int p4_devport_mgr_pcie_cpu_port_get(int dev);`

```suggestion
  return CHECK_RETURN_IF_FALSE(p4_devport_mgr_pcie_cpu_port_get(unit) == 0);
```

Reply:
You are right, it will return `-1` if it cannot find the correct CPU port